rafthttp/stream_test.go

```diff
@@ -59,11 +59,9 @@ func TestStreamWriterAttachOutgoingConn(t *testing.T) {
 			}
 		}
 
-		// msgc has been swapped with a new one now that prevwfc is closed
-		msgc, ok := sw.writec()
-		if !ok {
-			t.Errorf("#%d: working status = %v, want true", i, ok)
-		}
+		// msgc has either been swapped with a new one after closing prevwfc or
+		// is the initial msgc (in which case the first connection may be pending)
+		msgc, _ := sw.writec()
```

Comment:
it won't get the old msgc, `prevwfc.closed` is closed and msgc is updated in the same critical section; if `prevwfc.closed` is closed then `writec` can only acquire the stream writer lock after msgc is updated